.github/workflows/benchmark.yml

```
@@ -19,21 +19,19 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
   cancel-in-progress: true
 
+permissions:
+  contents: read
+
 jobs:
   benchmarks:
-    runs-on: buildjet-4vcpu-ubuntu-2204
+    runs-on: warp-ubuntu-latest-x64-4x
     steps:
       - uses: actions/checkout@v4
       - uses: dtolnay/rust-toolchain@stable
-      - uses: buildjet/cache@v3
+      - uses: Swatinem/rust-cache@v2
```

We should use warp builds cache actions for rust here to avoid the slow GHA cache.

https://docs.warpbuild.com/ci/cache/quickstart#rust

```suggestion
      - uses: WarpBuilds/rust-cache@v2
```